src/simulator_core/entities/esdl_object.py

```diff
@@ -51,3 +53,30 @@ def get_all_assets_of_type(self, esdl_asset_type: str) -> list[EsdlAssetObject]:
         return [EsdlAssetObject(asset)
                 for asset in self.energy_system_handler.get_all_instances_of_type(
                 StringEsdlAssetMapper().to_esdl(esdl_asset_type))]
+
+    def get_connected_assets(self, id: str, port: Port) -> List[Tuple[str, Port]]:
+        """Method to get the id's of connected assets from the esdl.
+
+        This returns a list of list with the connected asset id and the port to which it is
+        connected to the asset.
+
+        :param str id: id of the asset for which we want to know the connected assets
+        :param Port port: port for which the connected assets need to be returned.
+        :return: List of list which the id of the connected assets and the connected port.
+        """
+        esdl_asset = self.energy_system_handler.get_by_id(id)
+        connected_assets = []
+        type_port = esdl.InPort
+        if port == Port.Out:
+            type_port = esdl.OutPort
+        connected_port_ids = []
+        for esdl_port in esdl_asset.port:
+            if isinstance(esdl_port, type_port):
+                connected_port_ids = esdl_port.connectedTo
```

Comment:
This seems odd. You're initializing a list and then overwriting the list with a single value. I believe you would like to append all connected ports to the "connected_port_ids" list. 

I also do not understand why you would like to break this statement. 

Can you combine the first loop with the second loop? You can directly assign the type of the connected asset. 

I would also make a separate method for the type check of the port. You are using it between line 69 and 71 and at this location.

Reply:
Not entirely if the asset does not have the requested type of port then the list would not exist and the next step would fail.